books/data_transformation.py

```diff
@@ -1,108 +1,128 @@
+        ])
+
+        new_api_users_column_names = {
+            'location.city': 'user_city',
+            'location.country': 'user_country',
+            'registered.age': 'user_age'
+        }
+        df_users.rename(columns=new_api_users_column_names, inplace=True)
+        df_users['user_age'] = df_users['user_age'].fillna(0).astype(int)
+        self.data_frames['api_users'] = df_users
+
+    def transform_scrapped_books_data(self, source_path):
+        scrapped_data = pd.read_csv(os.path.join(source_path, 'scrapped_book_data.csv'), delimiter=';', usecols=['Title', 'Author', 'ISBN/ISSN:', 'Wydawnictwo:', 'Rok wydania:'], low_memory=False)
+        new_scraped_data_column_names = {
+            'ISBN/ISSN:': 'book_isbn',
+            'Title': 'book_title',
+            'Author': 'book_author',
+            'Rok wydania:': 'year_published',
+            'Wydawnictwo:': 'publisher',
+        }
+        scrapped_data.rename(columns=new_scraped_data_column_names, inplace=True)
+        scrapped_data['year_published'] = scrapped_data['year_published'].fillna(0).astype(int)
+        self.data_frames['scrapped_data'] = scrapped_data
+
+    def save_to_csv(self, data_type, target_path):
+        df = self.data_frames.get(data_type)
+        if df is not None:
+            file_name = os.path.join(target_path, f"transformed_{data_type}_file_{datetime.now().strftime('%Y_%m_%d_%H_%M_%S')}.csv")
+            df.to_csv(file_name, index=False)
+            logging.info(f"{data_type.capitalize()} data saved to {file_name}")
+        else:
+            logging.error(f"No data found for type: {data_type}")
 
 
 if __name__ == '__main__':
-    transform_books()
-    transform_users()
-    transform_ratings()
-    transform_user_api_data()
-    transform_scrapped_books_data()
+    parser = ArgumentParser(prog='Data Transformation')
+    parser.add_argument('-t', '--target_path', default='./_data/books/transformed', type=str)
+    parser.add_argument('-s', '--source_path', default='./_data/books/raw', type=str)
+    args = parser.parse_args()
+    transformation = DataTransformation()
+    transformation.transform_books(source_path=args.source_path)
+    transformation.transform_users(source_path=args.source_path)
+    transformation.transform_ratings(source_path=args.source_path)
+    # transformation.transform_user_api_data(source_path=args.source_path)
+    # transformation.transform_scrapped_books_data(source_path=args.source_path) >I think we should delete fun save to csv from scrapper_books.py in order to have transformation and saving files in one place
+    transformation.save_to_csv('books', target_path=args.target_path)
+    transformation.save_to_csv('users', target_path=args.target_path)
+    transformation.save_to_csv('ratings', target_path=args.target_path)
+    # transformation.save_to_csv('api_users', target_path=args.target_path)
+    # python books/data_transformation.py -t './_data/books/transformed' -s './_data/books/raw'
```

Comment:
This is fine but..... If I were writing this script I would do it in a way that:
```python
transformation = DataTransformation()
transformation.transform_books(source_path=args.source_path, target_path=args.target_path)
```

and call only a single method to transform data :)

Reply:
Yes exactly!